test/pkg/environment/expectations.go

```diff
@@ -0,0 +1,89 @@
+package environment
+
+import (
+	"sync"
+
+	. "github.com/onsi/gomega" //nolint:revive,stylecheck
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	storagev1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter/pkg/apis/awsnodetemplate/v1alpha1"
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+)
+
+var (
+	EnvironmentLabelName = "testing.karpenter.sh/environment-name"
+	CleanableObjects     = []client.Object{
+		&v1alpha5.Provisioner{},
+		&v1.Pod{},
+		&appsv1.Deployment{},
+		&appsv1.DaemonSet{},
+		&v1beta1.PodDisruptionBudget{},
+		&v1.PersistentVolumeClaim{},
+		&v1.PersistentVolume{},
+		&storagev1.StorageClass{},
+		&v1alpha1.AWSNodeTemplate{},
+	}
+)
+
+func (env *Environment) ExpectCreated(objects ...client.Object) {
+	for _, object := range objects {
+		object.SetLabels(map[string]string{EnvironmentLabelName: env.Options.EnvironmentName})
+		Expect(env.Client.Create(env, object)).To(Succeed())
+	}
+}
+
+func (env *Environment) ExpectCleaned() {
+	namespaces := &v1.NamespaceList{}
+	env.Expect(env.Client.List(env, namespaces)).To(Succeed())
+	wg := sync.WaitGroup{}
+	for _, object := range CleanableObjects {
+		for _, namespace := range namespaces.Items {
+			wg.Add(1)
+			go func(object client.Object, namespace string) {
+				env.Expect(env.Client.DeleteAllOf(env, object,
```

Comment:
Maybe add `&client.DeleteAllOfOptions{DeleteOptions: client.DeleteOptions{GracePeriodSeconds: ptr.Int64(0)}})`

Reply:
I wanted this to be closer to natural usage of clusters and include normal deletion flows rather than force deletion.